lib/Targets/AIETargetAirbin.cpp

```diff
@@ -0,0 +1,1316 @@
+//===- AIETargetAirbin.cpp --------------------------------------*- C++ -*-===//
+//
+// This file is licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// (c) Copyright 2021 Xilinx Inc.
+//
+//===----------------------------------------------------------------------===//
+
+#include "aie/Dialect/AIE/IR/AIEDialect.h"
+#include "aie/Dialect/AIE/IR/AIETargetModel.h"
+#include "aie/Dialect/AIEX/IR/AIEXDialect.h"
+
+#include "llvm/Support/Debug.h"
+#include "llvm/Support/Format.h"
+#include "llvm/Support/FormatVariadic.h"
+#include "llvm/Support/raw_ostream.h"
+
+#include <fcntl.h> // open
+#include <gelf.h>
+#include <iostream>
+#include <libelf.h>
+#include <sys/stat.h>
+#include <unistd.h> // read
+#include <utility>  // pair
+#include <vector>
+
+#define DEBUG_TYPE "aie-generate-airbin"
+
+#define EM_AMDAIR 225 /* AMD AIR */
```

Comment:
the team agreed - this is the way to go
we are not interested in defining our own official ELF format at this time, so we will just stick with this define
This will likely be moved to a header file in the near future, but for now this is fine